mbuild/lattice.py

```diff
@@ -15,68 +14,66 @@ class Lattice(object):
+
 
     Parameters
     ----------
-    dimension : int, optional, default=3
-        Dimension of the system of interest.
-    lattice_vectors : numpy array, shape=(dimension, dimension), optional
-                      default=([1,0,0], [0,1,0], [0,0,1])
-        Vectors that define edges of unit cell corresponding to dimension.
-    lattice_spacings : list-like, shape=(dimension,), optional, default=None
-        Length of unit cell edges.
-    basis_atoms : dictionary, shape={'id':[nested list of coordinate pairs]}
-                    default={'default':[[0., 0., 0.]]
-        Location of all basis Compounds in unit cell.
-    angles : list-like,  shape=(dimension,), optional, default=None
-        Interplanar angles describing unit cell.
+    lattice_spacing : numpy array, shape=(3,), required, dtype=float
+        Array of lattice spacings a,b,c for the cell.
+    lattice_vectors : numpy array, shape=(3, 3), optional
+                      default=[[1,0,0], [0,1,0], [0,0,1]]
+        Vectors that encase the unit cell corresponding to dimension. Will
+        only default to these values if no angles were defined as well.
+    lattice_points : dictionary, shape={'id': [[nested list of positions]]
+        optional, default={'default': [[0.,0.,0.]]}
+        Locations of all lattice points in cell using fractional coordinates.
+    angles : numpy array, shape=(3,), optional, dtype=float
+        Array of inter-planar Bravais angles
 
     Attributes
     ----------
-    dimension : int, optional, default=3
-        Dimension of system of interest
-    lattice_vectors : numpy array, shape=(dimension, dimension), optional
-                      default=([1,0,0], [0,1,0], [0,0,1])
-        Vectors that define edges of unit cell corresponding to dimension.
-    lattice_spacings : list-like, shape=(dimension,), required, default=None
-        Length of unit cell edges.
-    basis_atoms : list-like, shape=(['id',[dimension,]], ... ,) optional
-                    default={('default',([0,0,0]))}
-        Location of all basis Compounds in unit cell.
-    angles : list-like, optional, default=None
-        Lattice angles to define Bravais Lattice.
+    dimension : int, 3
+        Default dimensionality within mBuild. If choosing a lower dimension,
+        pad the relevant arrays with zeroes.
+    lattice_spacing : numpy array, shape=(3,), required, dtype=float
+        Array of lattice spacings a,b,c for the cell.
+    lattice_vectors : numpy array, shape=(3, 3), optional
+                      default=[[1,0,0], [0,1,0], [0,0,1]]
+        Vectors that encase the unit cell corresponding to dimension. Will
+        only default to these values if no angles were defined as well.
+    lattice_points : dictionary, shape={'id': [[nested list of positions]]
+        optional, default={'default': [[0.,0.,0.]]}
+        Locations of all lattice points in cell using fractional coordinates.
+    angles : numpy array, shape=(3,), optional, dtype=float
```

Comment:
Was there a reason this was changed from `list-like` to `numpy array`? It looks like lists are also supported and converted to numpy arrays in `_validate_angles`.

Reply:
Wasnt sure what the best way to describe it should be. It can take anything as long as it can be converted to a proper `numpy array`. 

However, under the hood, it is converted to an array immediately. 

I guess `list-like` would be the proper terminology.

Reply:
`array_like` is the most accurate, since anything that is `array_like` can be converted into an `ndarray`